ffi.md

```diff
@@ -4,6 +4,97 @@ title: Foreign Function Interfaces
 
 The ANSI Common Lisp standard doesn't mention this topic. So almost everything that can be said here depends on your OS and your implementation.
 
+### Example: Calling mathematical functions from C's `math.h` header file.
+ 
+Let's use `defcfun` to interface with the foreign [ceil](https://en.cppreference.com/w/c/numeric/math/ceil) C function from `math.h`.
+
+[defcfun](https://cffi.common-lisp.dev/manual/html_node/defcfun.html) is a macro in the cffi library that generates a function with the name you give it in your local scope.
+
+CL-USER> CL-USER> (cffi:defcfun ("ceil" c-ceil) :double (number :double))
+
+Here is the above function macroexpanded with `macrostep-expand`:
+
+~~~lisp
+(progn
+	  nil
+	  (defun c-ceil (number)
+	    (let ((#:g312 number))
+	      (cffi-sys:%foreign-funcall "ceil" (:double #:g312 :double) :convention
+					 :cdecl :library :default))))
+~~~
+
+As we can see from the above macroexpansion, and ignoring other details, a function is created with the builtin `defun` that is called `c-ceil` and that takes exactly one argument: `number`.
```

Comment:
We don't usually refer to "builtins" in Lisp, and really I don't think you need to explain what `defun` is. I would say "a function `c-ceil` is defined that takes exactly one argument"